tensorflow/python/training/learning_rate_decay_test.py

```diff
@@ -29,6 +29,68 @@
 from tensorflow.python.training import learning_rate_decay
 
 
+class CyclicLearningRateTest(test_util.TensorFlowTestCase):
+  """Functional test for Cyclic Learning Rate"""
+  
+  def np_cyclic_learning_rate(self, step, lr, max_lr, step_size, mode):
+    cycle = math.floor(1. + step / (2. * step_size))
+    x = math.fabs(step / step_size - 2. * cycle + 1.)
+    clr = (max_lr - lr) * max(0., 1. - x)
```

Comment:
Different than https://github.com/bckenstler/CLR
cycle = np.floor(1+step/(2*step_size))
x = np.abs(step/step_size - 2*cycle + 1)
clr = base_lr + (max_lr-base_lr)*np.maximum(0, (1-x))

Reply:
Thanks @sguada,
Sorry I didn't catch the difference, are you referring to the usage of Numpy instead of python's math module?
For the last statement: "`clr = base_lr + (max_lr-base_lr)*np.maximum(0, (1-x))`", I am delaying the addition of `base_lr` till the end, after handling all possible modes.